### PR TITLE
near-vm: restore near-vm tests

### PR DESCRIPTION
**Pull request by nagisa**
This makes sure we run all of the tests we have for the runtime. I don't particularly enjoy or appreciate the compexity of the test suite setup (e.g. we no longer need the proc macro since we only have and will ever have a single engine and compiler...), but I felt like it'd be better to restore the original functionality first and then simplify later.

### CODE REVIEW

File: Cargo.toml
Comment (Ekleog-NEAR) [inline Cargo.toml:140]: Is the reordering of almost all crates that include a dash due to some tooling that we could automate and lint on, to avoid merge conflicts from now on? If not I’d think keeping the current order would make sense.

File: runtime/near-vm-runner/src/near_vm_runner.rs
Comment (Ekleog-NEAR) [inline runtime/near-vm-runner/src/near_vm_runner.rs:240]: Uhhh reading this I feel like I misunderstood the scope of this PR, so I’d be reviewing it wrong if I were to continue (more tests should not lead to compilation changes).

Stopping my review here, let me know when I should review again :)

File: runtime/near-vm-runner/src/wasmer2_runner.rs
Comment (Ekleog-NEAR) [inline runtime/near-vm-runner/src/wasmer2_runner.rs:236]: (also I don’t really get why wasmer2 also needs a bump, if this is only about near-vm)

File: runtime/near-vm-runner/src/near_vm_runner.rs
Comment (nagisa) [reply runtime/near-vm-runner/src/near_vm_runner.rs:240]: This PR was originally based on the PR that upgraded the toolchain.

File: Cargo.toml
Comment (nagisa) [reply Cargo.toml:140]: I’m honestly too lazy with maintaining the order manually. If this is not a-ok, I’m inclined to just add anything at the end of the dependency list (making this sorted does not really help anything, when you have grep equivalents for searching)

Comment (Ekleog-NEAR) [reply Cargo.toml:140]: I’m entirely with you about maintaining the order manually, but then I’m curious how you reordered the things / if there’s a tool we could have as a pre-commit-hook like, to avoid such large changes happening too often?

Comment (nagisa) [reply Cargo.toml:140]: What I did was basically a (in vim)

```
(go to the top of the list)
^V
}
:!sort
```

so unfortunately that doesn't seem to be readily automatable. That said, there are some suggestions for automated tools in https://github.com/rust-lang/rustfmt/issues/4091…

**Review by Ekleog-NEAR — APPROVED** [review]
LGTM! I’m just not sure about that `publish = false` change, so removing the automerge label before approving so you have a chance to look at it :)

**Review by Ekleog-NEAR — APPROVED** [review]
LGTM! I’m just not sure about that `publish = false` change, so removing the automerge label before approving so you have a chance to look at it :)

**Comment by Ekleog-NEAR** [comment]
Oh and I forgot to write down: I’m not entirely sure how it restores the tests, but I completely see how it fixes https://github.com/near/nearcore/issues/8834 :)

**Comment by nagisa** [comment]
`near-vm/api` crate is *the* top level crate from the original wasmer project. We do not use it in the runtime implementation for a long time now, as it does not add all that much of value. It is still being depended on by tests though.

The tests get restored by `near-vm/api` taking the role of being the entry point for executing these tests now. I moved the `near-vm/build.rs` to `near-vm/api/build.rs` – this makes the test generator to actually get run, and the tests then also start running.

File: runtime/near-vm/api/src/sys/externals/table.rs
Comment (nagisa) [reply runtime/near-vm/api/src/sys/externals/table.rs:80]: Will fix in a follow-up.

**Pull request merged**
